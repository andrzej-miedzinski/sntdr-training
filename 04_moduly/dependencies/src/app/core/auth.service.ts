import { Injectable } from '@angular/core';

@Injectable({
  providedIn: 'root'
})
export class AuthService {
  login() {
    console.log('Użytkownik zalogowany.');
  }

  logout() {
    console.log('Użytkownik wylogowany.');
  }
}