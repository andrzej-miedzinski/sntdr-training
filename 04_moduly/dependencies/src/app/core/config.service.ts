import { Injectable } from '@angular/core';

@Injectable({
  providedIn: 'root'
})
export class ConfigService {
  getConfig() {
    return { apiUrl: 'https://api.example.com' };
  }
}