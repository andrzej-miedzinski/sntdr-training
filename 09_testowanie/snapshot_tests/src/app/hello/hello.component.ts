import { Component } from '@angular/core';

@Component({
  selector: 'app-hello',
  template: `<p>Hello, World!</p>`
})
export class HelloComponent {}