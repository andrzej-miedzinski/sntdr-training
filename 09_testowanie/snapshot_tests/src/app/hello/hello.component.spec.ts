import { TestBed, ComponentFixture } from '@angular/core/testing';
import { HelloComponent } from './hello.component';

describe('HelloComponent', () => {
  let component: HelloComponent;
  let fixture: ComponentFixture<HelloComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [HelloComponent]
    });
    fixture = TestBed.createComponent(HelloComponent);
    component = fixture.componentInstance;
    fixture.detectChanges(); // Renderuje komponent
  });

  it('powinien pasowaÄ‡ do snapshotu', () => {
    expect(fixture.nativeElement.outerHTML).toMatchSnapshot();
  });
});