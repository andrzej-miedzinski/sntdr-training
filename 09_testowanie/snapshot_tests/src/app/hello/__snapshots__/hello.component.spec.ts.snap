// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloComponent powinien pasowaÄ‡ do snapshotu 1`] = `"<div id="root0" ng-version="18.2.13"><p>Hello, World!</p></div>"`;
